src/core/asyncUtils.ts

```diff
@@ -20,11 +20,11 @@ function asyncUtils(act: Act, addResolver: (callback: () => void) => void): Asyn
       return callbackResult ?? callbackResult === undefined
     }
 
-    const timeoutSignal = createTimeoutController(timeout)
+    const timeoutSignal = createTimeoutController(timeout as number | boolean, false)
 
     const waitForResult = async () => {
       while (true) {
-        const intervalSignal = createTimeoutController(interval)
+        const intervalSignal = createTimeoutController(interval as number | boolean, true)
         timeoutSignal.onTimeout(() => intervalSignal.cancel())
 
         await intervalSignal.wrap(new Promise<void>(addResolver))
```

Comment:
I've just noticed that I didn't rename these to `timeoutController` and `intervalController` when I renamed `createTimeoutSignal` to `createTimeoutController` 😅

Reply:
Yeah, 'intervalController' and 'timeoutController' are a better names , I will change them
👍